catch/unit/texture/hipTextureObjectTests.cc

```diff
@@ -0,0 +1,220 @@
+#if HT_AMD
+      res_vew_desc.format = hipResViewFormatFloat1;
+      res_vew_desc.width = size;
+#else
+      std::cout << "Resource View Descriptors are not supported on NVIDIA currently" << std::endl;
+      useResourceViewDescriptor = false;
+#endif
+    }
+
+
+    HIP_CHECK(hipCreateTextureObject(&texture_object, &res_desc, &tex_desc,
+                                     useResourceViewDescriptor ? &res_vew_desc : nullptr));
+  }
+
+  ~TextureObjectTestWrapper() {
+    if (!ommit_destroy_) {
+      HIP_CHECK(hipDestroyTextureObject(texture_object));
+    }
+    HIP_CHECK(hipFreeArray(array_member));
+    free(host_data_);
+  }
+};
+
+/* hipGetTextureObjectResourceDesc tests */
+
+TEST_CASE("Unit_hipGetTextureObjectResourceDesc_positive") {
+  CHECK_IMAGE_SUPPORT;
+
+  TextureObjectTestWrapper tex_obj_wrapper(false);
+
+  hipResourceDesc check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipGetTextureObjectResourceDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.resType == tex_obj_wrapper.res_desc.resType);
+  REQUIRE(check_desc.res.array.array == tex_obj_wrapper.res_desc.res.array.array);
+}
+
+
+TEST_CASE("Unit_hipGetTextureObjectResourceDesc_Negative_Parameters") {
+  CHECK_IMAGE_SUPPORT;
+
+  TextureObjectTestWrapper tex_obj_wrapper(false);
+
+  hipResourceDesc check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  SECTION("desc is nullptr") {
+    HIP_CHECK_ERROR(hipGetTextureObjectResourceDesc(nullptr, tex_obj_wrapper.texture_object),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("texture is invalid") {
+    HIP_CHECK_ERROR(
+        hipGetTextureObjectResourceDesc(&check_desc, static_cast<hipTextureObject_t>(0)),
+        hipErrorInvalidValue);
+  }
+}
+
+/* hipGetTextureObjectResourceViewDesc tests */
+
+#if HT_AMD
+TEST_CASE("Unit_hipGetTextureObjectResourceViewDesc_positive") {
+  CHECK_IMAGE_SUPPORT;
+
+  TextureObjectTestWrapper tex_obj_wrapper(true);
+
+  hipResourceViewDesc check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipGetTextureObjectResourceViewDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.format == tex_obj_wrapper.res_vew_desc.format);
+  REQUIRE(check_desc.width == tex_obj_wrapper.res_vew_desc.width);
+}
+#endif
+
+#if HT_AMD
+TEST_CASE("Unit_hipGetTextureObjectResourceViewDesc_Negative_Parameters") {
+  CHECK_IMAGE_SUPPORT;
+
+  TextureObjectTestWrapper tex_obj_wrapper(true);
+
+  hipResourceViewDesc check_desc;
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  SECTION("desc is nullptr") {
+    HIP_CHECK_ERROR(hipGetTextureObjectResourceViewDesc(nullptr, tex_obj_wrapper.texture_object),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("texture is invalid") {
+    HIP_CHECK_ERROR(
+        hipGetTextureObjectResourceViewDesc(&check_desc, static_cast<hipTextureObject_t>(0)),
+        hipErrorInvalidValue);
+  }
+
+  HipTest::HIP_SKIP_TEST("Skipping on NVIDIA platform");
```

Comment:
you don't need this here then?

Reply:
@chrispaquot Removed skipping of nvidia test that is disabled.

Reply:
@rakesroy It seems all changes made in this commit have been already merged as part of https://github.com/ROCm/hip-tests/pull/92 . When you verify this, the PR can be closed.